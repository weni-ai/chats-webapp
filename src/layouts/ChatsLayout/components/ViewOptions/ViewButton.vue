<template>
  <section class="view-button-content">
    <UnnnicButton
      size="large"
      type="tertiary"
      class="view-button"
      @click="handleClick"
    >
      <p class="view-button__text">{{ $t('view-option') }}</p>
      <UnnnicIcon
        data-testid="header-icon-expand"
        size="md"
        :icon="expandedMore ? 'expand_more' : 'expand_less'"
        scheme="neutral-darkest"
      />
    </UnnnicButton>
  </section>
</template>

<script setup>
import { defineProps } from 'vue';

defineProps({
  expandedMore: Boolean,
  handleClick: {
    type: Function,
    default: null,
  },
});
</script>

<style lang="scss" scoped>
.view-button-content {
  display: flex;
  align-items: center;

  :deep(.unnnic-button) {
    width: 100%;
    display: flex;
    justify-content: space-between;
    padding: $unnnic-spacing-sm;
    align-self: stretch;
    border-top: 1px solid $unnnic-color-neutral-soft;
    border-bottom: 1px solid $unnnic-color-neutral-soft;
    border-left: 1px solid $unnnic-color-neutral-soft;
    background-color: $unnnic-color-neutral-lightest;
  }

  :deep(.unnnic-button--tertiary) {
    background-color: $unnnic-color-neutral-lightest;
  }

  :deep(.unnnic-button--tertiary:active:enabled) {
    background-color: $unnnic-color-neutral-lightest;
  }

  :deep(.unnnic-button--tertiary:hover:enabled) {
    background-color: $unnnic-color-neutral-lightest;
    border-top: 1px solid $unnnic-color-neutral-cleanest;
    border-bottom: 1px solid $unnnic-color-neutral-cleanest;
    border-left: 1px solid $unnnic-color-neutral-cleanest;
  }
}

.view-button {
  :deep(.unnnic-button__label) {
    width: 100%;
    display: flex;
    justify-content: space-between;
  }

  &__text {
    color: $unnnic-color-neutral-cloudy;

    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-gt;
    font-style: normal;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }
}
</style>
