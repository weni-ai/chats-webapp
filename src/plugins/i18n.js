import Vue from 'vue';
import VueI18n from 'vue-i18n';

// eslint-disable-next-line camelcase
import pt_br from '@/locales/pt_br.json';
import en from '@/locales/en.json';
import es from '@/locales/es.json';

import moment from 'moment';

import('moment/locale/pt-br');
import('moment/locale/es');

moment.locale('pt-br');

Vue.use(VueI18n);

const messages = {
  'pt-br': pt_br,
  en,
  es,
};

const i18n = new VueI18n({
  locale: 'pt-br',
  fallbackLocale: 'pt-br',
  messages,
  dateTimeFormats: {
    'pt-br': {
      short: {
        year: 'numeric',
        month: '2-digit',
        day: '2-digit',
      },
      long: {
        month: 'long',
        day: '2-digit',
      },
    },
    en: {
      short: {
        year: 'numeric',
        month: '2-digit',
        day: '2-digit',
      },
      long: {
        month: 'long',
        day: '2-digit',
      },
    },
    es: {
      short: {
        year: 'numeric',
        month: '2-digit',
        day: '2-digit',
      },
      long: {
        month: 'long',
        day: '2-digit',
      },
    },
  },
});

export default i18n;
