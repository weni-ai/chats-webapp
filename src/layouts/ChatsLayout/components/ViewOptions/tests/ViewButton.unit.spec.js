import { afterAll, beforeAll, describe, expect, it, vi } from 'vitest';
import { mount, config } from '@vue/test-utils';
import ViewButton from '@/layouts/ChatsLayout/components/ViewOptions/ViewButton.vue';
import i18n from '@/plugins/i18n';

beforeAll(() => {
  config.global.plugins = config.global.plugins.filter(
    (plugin) => plugin !== i18n,
  );
});

afterAll(() => {
  if (!config.global.plugins.includes(i18n)) {
    config.global.plugins.push(i18n);
  }
});

describe('ViewButton', () => {
  let wrapper;

  const createWrapper = (props) => {
    return mount(ViewButton, {
      props,
      global: {
        mocks: {
          $t: (key) => key,
        },
        stubs: {
          UnnnicButton: {
            template: '<button><slot /></button>',
          },
          UnnnicIcon: {
            template: '<span data-testid="icon" :data-icon="icon"></span>',
            props: ['icon'],
          },
        },
      },
    });
  };

  describe('Props', () => {
    it('should receive expandedMore as true', () => {
      wrapper = createWrapper({ expandedMore: true });
      expect(wrapper.props('expandedMore')).toBe(true);
    });

    it('should receive expandedMore as false', () => {
      wrapper = createWrapper({ expandedMore: false });
      expect(wrapper.props('expandedMore')).toBe(false);
    });
  });

  describe('Rendering Content', () => {
    it('should display translated text inside the button', () => {
      wrapper = createWrapper({ expandedMore: true });
      expect(wrapper.find('[data-testid="header-text"]').text()).toBe(
        'view-option',
      );
    });

    it('should display correct icon when expandedMore is true', () => {
      wrapper = createWrapper({ expandedMore: true });
      const icon = wrapper.find('[data-testid="header-icon-expand"]');
      expect(icon.attributes('data-icon')).toBe('expand_more');
    });

    it('should display correct icon when expandedMore is false', () => {
      wrapper = createWrapper({ expandedMore: false });
      const icon = wrapper.find('[data-testid="header-icon-expand"]');
      expect(icon.attributes('data-icon')).toBe('expand_less');
    });
  });

  describe('Events', () => {
    it('should call handleClick when button is clicked', async () => {
      const handleClick = vi.fn();
      wrapper = createWrapper({ expandedMore: true, handleClick });
      await wrapper.find('button').trigger('click');
      expect(handleClick).toHaveBeenCalled();
    });
  });

  describe('Snapshot', () => {
    it('should match the snapshot', () => {
      wrapper = createWrapper({ expandedMore: true });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
