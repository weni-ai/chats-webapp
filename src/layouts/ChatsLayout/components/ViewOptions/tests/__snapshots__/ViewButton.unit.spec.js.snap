// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ViewButton > Snapshot > should match the snapshot 1`] = `
<section
  class="view-button-content"
  data-v-905ee5f6=""
>
  <button
    class="view-button"
    data-testid="header-btn"
    data-v-905ee5f6=""
    size="large"
    type="submit"
  >
    
    <p
      class="view-button__text"
      data-testid="header-text"
      data-v-905ee5f6=""
    >
      view-option
    </p>
    <span
      class="material-symbols-rounded unnnic-icon-scheme--neutral-darkest unnnic-icon-size--md"
      data-testid="header-icon-expand"
      data-v-905ee5f6=""
      data-v-9846e6db=""
      translate="no"
    >
      expand_more
    </span>
    
  </button>
</section>
`;
